src/main/webapp/event-controller.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** Manages storing eventListeners and broadcasting events. */
+export default class EventController {
+  constructor() {
+    this.eventListeners = new Map();
+    this.addEventListener = this.addEventListener.bind(this);
+    this.broadcastEvent = this.broadcastEvent.bind(this);
```

Comment:
Why are these here?  There are only a few situations where you need to bind functions now that we have classes

Reply:
Classes aren't in this branch yet, but I removed for the future